evaluate/task/task-transpile.go

```diff
@@ -170,49 +162,66 @@ func (t *TaskTranspile) Run(ctx evaltask.Context) (repositoryAssessment map[eval
 	return repositoryAssessment, problems, nil
 }
 
-// unpackTranspilerPackage checks if the testdata repository for the transpilation task is well-formed and returns the path to the implementation file and also the path to the file that holds the stub.
-func (t *TaskTranspile) unpackTranspilerPackage(ctx evaltask.Context, fileLogger *log.Logger, originLanguage language.Language, packagePath string) (originFilePath string, stubFilePath string, err error) {
+// unpackTranspilerPackage returns the path to the implementation file and also the path to the file that holds the stub.
+func (t *TaskTranspile) unpackTranspilerPackage(ctx evaltask.Context, fileLogger *log.Logger, packagePath string) (originFilePathsWithLanguage map[string]language.Language, stubFilePath string, err error) {
+	originFilePathsWithLanguage = map[string]language.Language{}
 	packagePathAbsolute := filepath.Join(ctx.Repository.DataPath(), packagePath)
 
-	files, err := originLanguage.Files(fileLogger, filepath.Join(packagePathAbsolute, "implementation"))
+	files, err := os.ReadDir(filepath.Join(packagePathAbsolute, "implementation"))
 	if err != nil {
-		return "", "", pkgerrors.WithStack(err)
+		return nil, "", pkgerrors.WithStack(err)
+	}
+
+	for _, file := range files {
+		var originLanguage language.Language
+		for _, l := range language.Languages {
+			if filepath.Ext(file.Name()) == l.DefaultFileExtension() {
+				originLanguage = l
+			}
+		}
```

Comment:
Add a comment that we always assume that we only encounter files of languages that we actually support. Because otherwise an error is raised during repository validation.